src/lib/ng-package/discover-packages.ts

```diff
@@ -10,8 +10,11 @@ import { globFiles } from '../utils/glob';
 const ngPackageSchemaJson = require('../../ng-package.schema.json');
 
 interface UserPackage {
-  packageJson: object;
-  ngPackageJson: object;
+  /** Values from the `package.json` file of this user package. */
+  packageJson: Record<string, any>;
+  /** NgPackageConfig for this user package. */
+  ngPackageJson: Record<string, any>;
```

Comment:
We could also set this to `NgPackageConfig` if we remove the generated interface from `.gitignore`.